python/revng/scripting/projects/daemon_project.py

```diff
@@ -0,0 +1,192 @@
+
+import yaml
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+from revng.model import Binary, dump_diff_to_string, load_model_from_string, make_diff
+
+from .project import Project
+
+
+class DaemonProject(Project):
+    def __init__(self, url: str):
+        self.transport = AIOHTTPTransport(url=f"{url}/graphql/")
+        self.client = Client(
+            transport=self.transport, fetch_schema_from_transport=True, execute_timeout=None
+        )
+        super().__init__()
+
+    def import_binary(self, input_binary_path: str) -> None:
+        query = gql(
+            """
+            mutation upload($file: Upload!) {
+                uploadFile(file: $file, container: "input")
+            }
+        """
+        )
+
+        with open(input_binary_path, "rb") as binary_file:
+            result = self.client.execute(
+                query, variable_values={"file": binary_file}, upload_files=True
+            )
+
+        if not result["uploadFile"]:
+            raise RuntimeError("File upload failed")
+
+    def get_artifact(self, artifact_name: str) -> bytes | str:
+        return self._get_artifact_internal(artifact_name)
+
+    def _get_artifact_internal(self, artifact_name: str, targets_class: List[Any] = []) -> bytes | str:
+        if not targets_class:
+            targets_class = self.model.Functions
+        targets = [tc.key() for tc in targets_class]
+
+        result = self._get_artifact(artifact_name, targets)
+        return self._get_result_mime(artifact_name, result)
+
+    def _get_artifact(self, artifact_name: str, targets_class: List[str] = []) -> bytes | str:
+        query = gql(
+            """
+            query($step: String!, $paths: String!, $index: BigInt!) {
+                produceArtifacts(step: $step, paths: $paths, index: $index) {
+                    ... on Produced {
+                        result
+                    }
+                }
+            }
+        """
+        )
+        variables = {
+            "step": artifact_name,
+            "paths": ",".join(targets_class),
+            "index": self._get_content_commit_index(),
+        }
+
+        response = self.client.execute(query, variable_values=variables)
+        return response["produceArtifacts"]["result"].encode()
+
+    def analyze(self, analysis_name: str, target_class: list[Any] = []) -> None:
+        if analysis_name in self._get_analyses_list_names():
+            return self.analyses_list(analysis_name)
+
+        container_input = self._get_analysis_inputs(analysis_name)
+        step = self._get_step_name(analysis_name)
+
+        variables = {
+            "step": step,
+            "analysis": analysis_name,
+            "index": self._get_content_commit_index(),
+        }
+
+        # NOTE: passing a single Function as `target_class` parameter
+        # causes the daemon (LLVM) to crash
+        if not target_class:
+            target_class = self.model.Functions
```

Comment:
This is a hack, let's first check if the bug can be fixed before implementing it